valgrind.supp

```diff
@@ -0,0 +1,965 @@
+   fun:_ZNK5boost5proto6detail18default_shift_leftINS_7phoenix12meta_grammarEE4implIRNS3_5actorINS0_7exprns_10basic_exprINS0_6tagns_3tag10shift_leftENS0_7argsns_5list2INS7_INS9_ISC_NSE_INS7_INS_3log11v2_mt_posix11expressions3aux25attribute_output_terminalINS7_INS9_ISC_NSE_INS7_INSJ_INS7_INS9_ISC_NSE_INS7_INSJ_INS7_INS9_ISC_NSE_INS7_INS9_ISC_NSE_INS7_INS9_ISC_NSE_INS7_INSJ_INS7_INS9_INSB_8terminalENSD_4termINS3_8argumentILi2EEEEELl0EEEEENS_10posix_time5ptimeENSG_16fallback_to_noneENSG_3aux14light_functionIFvRNSG_24basic_formatting_ostreamIcNSt3__111char_traitsIcEENSX_9allocatorIcEEEERKSS_EEEEEEENS7_INS9_ISK_NSL_IA7_cEELl0EEEEEEELl2EEEEENS7_INS9_ISK_NSL_IiEELl0EEEEEEELl2EEEEENS7_INS9_ISK_NSL_IA3_cEELl0EEEEEEELl2EEEEEN2ml4core7CLogger6ELevelEST_NSG_10to_log_funIvEEEEEENS7_INS9_ISK_NSL_IcEELl0EEEEEEELl2EEEEENSX_12basic_stringIcSZ_S11_EEST_S1Z_EEEES24_EELl2EEEEEiST_S1Z_EEEES24_EELl2EEEEENSH_15attribute_actorIS29_ST_NSH_3tag8smessageES7_EEEELl2EEEEERNS3_7vector3IPKS2Q_RKNSG_11record_viewERNSI_10stream_refIS12_EEEERKNS3_15default_actionsEEclES2R_S32_S35_
+   fun:_ZNK5boost7phoenix9evaluator4implIRNS0_5actorINS_5proto7exprns_10basic_exprINS4_6tagns_3tag10shift_leftENS4_7argsns_5list2INS3_INS6_IS9_NSB_INS3_INS_3log11v2_mt_posix11expressions3aux25attribute_output_terminalINS3_INS6_IS9_NSB_INS3_INSG_INS3_INS6_IS9_NSB_INS3_INSG_INS3_INS6_IS9_NSB_INS3_INS6_IS9_NSB_INS3_INS6_IS9_NSB_INS3_INSG_INS3_INS6_INS8_8terminalENSA_4termINS0_8argumentILi2EEEEELl0EEEEENS_10posix_time5ptimeENSD_16fallback_to_noneENSD_3aux14light_functionIFvRNSD_24basic_formatting_ostreamIcNSt3__111char_traitsIcEENSU_9allocatorIcEEEERKSP_EEEEEEENS3_INS6_ISH_NSI_IA7_cEELl0EEEEEEELl2EEEEENS3_INS6_ISH_NSI_IiEELl0EEEEEEELl2EEEEENS3_INS6_ISH_NSI_IA3_cEELl0EEEEEEELl2EEEEEN2ml4core7CLogger6ELevelESQ_NSD_10to_log_funIvEEEEEENS3_INS6_ISH_NSI_IcEELl0EEEEEEELl2EEEEENSU_12basic_stringIcSW_SY_EESQ_S1W_EEEES21_EELl2EEEEEiSQ_S1W_EEEES21_EELl2EEEEENSE_15attribute_actorIS26_SQ_NSE_3tag8smessageES3_EEEELl2EEEEERKNS0_7vector2IRNS0_7vector3IPKS2N_RKNSD_11record_viewERNSF_10stream_refISZ_EEEERKNS0_15default_actionsEEENS4_6envns_9empty_envEEclES2O_S36_RKS38_
+}
+{
+   <insert_a_suppression_name_here>
+   Memcheck:Value8
+   fun:_platform_memmove$VARIANT$Haswell
+   fun:__sfvwrite
+   fun:__vfprintf
+   fun:__v2printf
+   fun:_vsnprintf
+   fun:vsnprintf_l
+   fun:snprintf_l
+   fun:_ZNKSt3__17num_putIcNS_19ostreambuf_iteratorIcNS_11char_traitsIcEEEEE6do_putES4_RNS_8ios_baseEcd
+   fun:_ZNSt3__113basic_ostreamIcNS_11char_traitsIcEEElsEd
+   fun:_ZNSt3__1lsIdcNS_11char_traitsIcEEEERNS_13basic_ostreamIT0_T1_EES7_RKNS_7complexIT_EE
+   fun:_ZN2ml4core14printer_detail12CLeafPrinterIN5boost17integral_constantIbLb0EEEE6print_INSt3__17complexIdEEEENS8_12basic_stringIcNS8_11char_traitsIcEENS8_9allocatorIcEEEERKT_S5_
+   fun:_ZN2ml4core14printer_detail12CLeafPrinterIN5boost17integral_constantIbLb0EEEE5printINSt3__17complexIdEEEENS8_12basic_stringIcNS8_11char_traitsIcEENS8_9allocatorIcEEEERKT_
```

Comment:
It's probably possible to reduce the number of suppressions by chopping some of the functions out of the stacks.  For example, it seems that on macOS Valgrind cannot properly trace the memory through the C++ runtime library and C runtime library when printing a number to an `ostream`.  So if you start the suppression here at `_ZNKSt3__17num_putIcNS_19ostreambuf_iteratorIcNS_11char_traitsIcEEEEE6do_putES4_RNS_8ios_baseEcd` and drop everything beneath it then this single suppression can probably cover several others, and they can be deleted.

Another advantage of this will be that similar suppressions won't have to be added each time somebody adds a new test that prints a number.